### PR TITLE
Add diffusion theory and associated kernels to Theory page on Moltres website

### LINKED ISSUE #267: Add documentation on neutron diffusion theory on the Moltres website
This issue can be closed when the [theory section of the Moltres website](https://arfc.github.io/moltres/getting_started/theory.html) contains:
- a brief introduction to multigroup neutron diffusion theory
- A table or list showing which Moltres kernels/BCs correspond to which mathematical terms in the multigroup neutron diffusion and delayed neutron precursor equations

### PR DESCRIPTION
**Pull request by nglaser3**
## Summary of changes

Add brief introduction to multi-group diffusion in Moltres, add table of terms in multi group diffusion equation and their associated kernels, add table of terms in delayed neutron precursor equation and their associated kernels. 

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Required for Merging
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
  - [ ] CI tests pass

## Associated Issues and PRs

- Resolves #267 

## Associated Developers

- Dev: @smpark7 

## Checklist for Reviewers

Reviewers should use [this link](https://arfc.github.io/manual/guides/pull_requests) to get to the 
Review Checklist before they begin their review.

### CODE REVIEW

**Comment by smpark7** [comment]
Hey @nglaser3, great work here! Before I provide specific review comments, could you break up your paragraphs into multiple lines with a max limit of 100 characters per line? This will help with indicating which parts of the text my review comments are referring to.

You can leave an empty line between the paragraphs to maintain separation in the rendered html file.

**Comment by nglaser3** [comment]
@smpark7, Thank you! I have broken the paragraphs into sentences, and checked to make sure the formatting was still correct. Please let me know if there are any other pre-check changes you would like me to make!

File: doc/content/getting_started/theory.md [hunk 1]
```diff
@@ -16,7 +16,61 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
...
+
+The general neutron diffusion equation has exceedingly few analytical solutions and thus, for real-world problems, is solved with numerical methods. 
+A common method utilized is [Finite Element Method](https://mooseframework.inl.gov/help/faq/what_is_fem.html), which is the methodology employed in Moltres. 
+To solve the diffusion equation over complex geometries, Moltres discretizes over space, time, and energy. 
+Discretization through space is done with the use of volume meshes. 
+Discretization with respect to time is accomplished with discrete time-steps. 
+Finally, the energy spectrum of the scalar neutron flux is discretized into user defined bins, called energy groups. 
+The energy discretization of the neutron diffusion equation is called the multi-group neutron diffusion equation and is presented below. 
+Notably, there are two production terms of neutrons, the prompt fission source and delayed neutron precursor decay source. 
+The first term describes the neutrons immediately born from fission, and the second term describes the neutrons born from the radioactive decay of neutron-emitting radionuclides, commonly called delayed neutron precursors.
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:34]: Move these to the top of the second paragraph

File: doc/content/getting_started/theory.md [hunk 2]
```diff
@@ -16,7 +16,61 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
...
+This assumption yields an equation with 2 fewer phase spaces, but also has reduced fidelity when compared to the transport equation, particularly in regions in which the neutron flux has strong angular dependence. 
+A non-exhaustive list of regions in which the neutron flux strongly depends on angle is near material interfaces, external boundaries, strong absorbers, and weak scaterrers.
+
+
+
+The general neutron diffusion equation has exceedingly few analytical solutions and thus, for real-world problems, is solved with numerical methods. 
+A common method utilized is [Finite Element Method](https://mooseframework.inl.gov/help/faq/what_is_fem.html), which is the methodology employed in Moltres. 
+To solve the diffusion equation over complex geometries, Moltres discretizes over space, time, and energy. 
+Discretization through space is done with the use of volume meshes. 
+Discretization with respect to time is accomplished with discrete time-steps.
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:30]:
```suggestion
In FEM, the spatial domain is discretized into finite mesh elements while the time dependence is modeled through discrete time steps and time integration methods.
```

File: doc/content/getting_started/theory.md [hunk 3]
```diff
@@ -16,7 +16,61 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
...
+| [!eq](- \nabla \cdot D_g \nabla \phi_g) | [GroupDiffusion](https://arfc.github.io/moltres/source/kernels/GroupDiffusion.html) | Streaming term of energy group g |
+| [!eq](\Sigma^R_{g} \phi_g) | [SigmaR](https://arfc.github.io/moltres/source/kernels/SigmaR.html) | Removal from energy group g |
+| [!eq](\sum^G_{g \neq g'} {\Sigma^s_{g'\rightarrow g} \phi_{g'}}) | [InScatter](https://arfc.github.io/moltres/source/kernels/InScatter.html) | In-scattering into energy group g | 
+| [!eq](\chi^p_{g} \sum^G_{g'=1} {\left[1- \beta \right] \nu_{g'} \Sigma^f_{g'}\phi_{g'} } ) | [CoupledFissionKernel](https://arfc.github.io/moltres/source/kernels/CoupledFissionKernel.html) | Prompt fission neutron source |
+| [!eq](\chi^d_{g} \sum^I_i {\lambda_i C_i}) | [DelayedNeutronSource](https://arfc.github.io/moltres/source/kernels/DelayedNeutronSource.html) | Delayed fission neutron source |
+
+ 
+\\  
+
+The governing equation for the delayed neutron precursor population:
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:60]:
```suggestion
The governing equation for the delayed neutron precursor distribution is:
```

**Review by smpark7 — CHANGES_REQUESTED** [review]
Good work on writing up this draft! I included several comments for typos and to improve the general flow.

General comments:
- Remove spaces at the end of lines
- Limit line lengths to 100 characters
- Use MooseDoc-style links for the kernel links

File: doc/content/getting_started/theory.md [hunk 4]
```diff
@@ -16,7 +16,47 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
 
 ## Multigroup Neutron Diffusion
 
-More information to be added in future.
+The neutron diffusion equation is an approximation to the Boltzmann transport equation, and is derived by taking the zeroth and first moment with respect to $\hat\Omega$, the direction of neutron travel. 
+A full derivation of the diffusion equation will not be presented here for conciceness. Importantly, when taking the first moment of the transport equation the angular flux is assumed to be linearly anisotropic.
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:20]:
```suggestion
A full derivation of the diffusion equation will not be presented here for conciseness. Importantly, when taking the first moment of the transport equation the angular flux is assumed to be linearly anisotropic. 
```

File: doc/content/getting_started/theory.md [hunk 5]
```diff
@@ -16,7 +16,47 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
...
 
-More information to be added in future.
+The neutron diffusion equation is an approximation to the Boltzmann transport equation, and is derived by taking the zeroth and first moment with respect to $\hat\Omega$, the direction of neutron travel. 
+A full derivation of the diffusion equation will not be presented here for conciceness. Importantly, when taking the first moment of the transport equation the angular flux is assumed to be linearly anisotropic. 
+The approximations reduce the phase space through the elimination of angular dependence, but the resulting equation also has reduced fidelity when compared to the transport equation, particularly in regions where the neutron flux has strong angular dependence. 
+A non-exhaustive list of regions in which the neutron flux strongly depends on angle is near material interfaces between materials with highly dissimilar neutronic properties, within strong absorbers, and within near-void regions.
+As a dertiministic method, the neutron diffusion method also requires discretization of the continuous energy dependence into energy groups consisting of non-overlapping, finite energy ranges across the entire energy spectrum.
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:23]:
```suggestion
As a deterministic method, the neutron diffusion method also requires discretization of the continuous energy dependence into energy groups consisting of non-overlapping, finite energy ranges across the entire energy spectrum. 
```

File: doc/content/getting_started/theory.md [hunk 6]
```diff
@@ -16,7 +16,47 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
...
+The approximations reduce the phase space through the elimination of angular dependence, but the resulting equation also has reduced fidelity when compared to the transport equation, particularly in regions where the neutron flux has strong angular dependence. 
+A non-exhaustive list of regions in which the neutron flux strongly depends on angle is near material interfaces between materials with highly dissimilar neutronic properties, within strong absorbers, and within near-void regions.
+As a dertiministic method, the neutron diffusion method also requires discretization of the continuous energy dependence into energy groups consisting of non-overlapping, finite energy ranges across the entire energy spectrum. 
+This energy discretization creates a system of equations referred to as the multigroup neutron diffusion equations: 
+
+!equation
+\frac{1}{v_g} \frac{\partial\phi_g}{\partial t}-\nabla \cdot D_g \nabla \phi_g +\Sigma^R_{g} \phi_g =\sum^G_{g \neq g'} {\Sigma^s_{g'\rightarrow g} \phi_{g'}}+ \chi^p_{g} \sum^G_{g'=1} {\left(1- \beta \right) \nu_{g'} \Sigma^f_{g'}\phi_{g'} }+\chi^d_{g} \sum^I_i {\lambda_i C_i}
+
+where the precursor distributions are governed by:
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:29]:
```suggestion
The delayed neutron precursor distributions are governed by:
```

File: doc/content/getting_started/theory.md [hunk 7]
```diff
@@ -16,7 +16,47 @@ the MOOSE finite element framework, enabling highly flexible and scalable reacto
...
+
+!equation
+\frac{\partial C_i}{\partial t}=\sum^G_{g'=1}{\beta_i \nu \Sigma^f_{g'}\phi_{g'}}-\lambda_i C_i-\vec{u} \cdot \nabla C_i
+
+
```
Comment (smpark7) [inline doc/content/getting_started/theory.md:34]:
```suggestion
```
Remove extra empty line here

**Review by smpark7 — CHANGES_REQUESTED** [review]
It looks good overall. I noted a few more corrections to be made.

**Review by smpark7 — APPROVED** [review]
Good work Nathan!

**Comment by smpark7** [comment]
@katyhuff @munkm Do you have any comments to add?

**Comment by katyhuff** [comment]
100 characters per line is still pretty long. Usually we go with 80. @smpark7 is there a reason for 100?

**Pull request merged**